packages/forklift-console-plugin/src/modules/Providers/utils/components/DetailsPage/DetailItem.tsx

```diff
@@ -132,55 +162,53 @@ export const DescriptionTitle: React.FC<{ title: string }> = ({ title }) => (
+ * The content field's element and if editable, next to that appears a press-able inline
  * link edit button with the pencil icon, for triggering the onEdit callback.
  *
  * @component
  * @param {ReactNode} content - The field's content element.
  * @param {Function} onEdit - Function to be called when the button is clicked.
  */
-export const EditableContentButton: React.FC<{
+
+export const ContentField: React.FC<{
   content: ReactNode;
   onEdit: () => void;
-}> = ({ content, onEdit }) => (
-  <DescriptionListDescription>
-    {content}
-    <Button variant="link" isInline onClick={onEdit}>
-      <Pencil className="forklift-page-details-edit-pencil" />
-    </Button>
-  </DescriptionListDescription>
-);
-
-/**
- * Component for displaying a non-editable content.
- *
- * @component
- * @param {ReactNode} content - The content of the description.
- */
-export const NonEditableContent: React.FC<{ content: ReactNode }> = ({ content }) => (
-  <DescriptionListDescription>{content}</DescriptionListDescription>
-);
+}> = ({ content, onEdit }) =>
+  onEdit ? (
+    <DescriptionListDescription>
+      {content}
+      <Button variant="link" isInline onClick={onEdit}>
+        <Pencil className="forklift-page-details-edit-pencil" />
+      </Button>
+    </DescriptionListDescription>
+  ) : (
+    <DescriptionListDescription>{content}</DescriptionListDescription>
+  );
 
 /**
  * Type for the props of the DetailsItem component.
  *
  * @typedef {Object} DetailsItemProps
  * @property {string} title - The title of the details item.
- * @property {ReactNode} content - The content of the details item.
  * @property {ReactNode} [helpContent] - The content to display in the help popover.
  * @property {ReactNode} [showHelpIconNextToTitle] - if true, adding a help icon next to the title for displaying the help popover.
  *    If false, show the default Patternfly dashed line under the title.
  * @property {string[]} [crumbs] - Breadcrumbs for the details item.
- * @property {Function} [onEdit] - Function to be called when the edit button is clicked.
+ * @property {ReactNode} content - first/only field content to be displayed for the details item.
+ * @property {ReactNode[]} contents - Array of content fields (starting from 2nd and forward) to be displayed for the details item.
+ * @property {Function[]} onEdit - function for the first/only content field, to be called when the edit button is clicked or null if the field is non editable.
+ * @property {Function[]} onEdits - Array of functions per content field (starting from 2nd and forward), to be called when the edit button is clicked or null if the field is non editable.
  */
 export type DetailsItemProps = {
   title: string;
-  content: ReactNode;
   helpContent?: ReactNode;
   showHelpIconNextToTitle?: boolean;
   moreInfoLabel?: string;
   moreInfoLink?: string;
   crumbs?: string[];
+  content: ReactNode;
+  contents?: ReactNode[];
   onEdit?: () => void;
+  onEdits?: (() => void)[];
```

Comment:
see comment https://github.com/kubev2v/forklift-console-plugin/pull/1135/files#r1597880185

```suggestion
  onEdit?: (() => void) | (() => void)[];
```